server/index.js

```diff
@@ -84,24 +88,104 @@ app.get('/test', (req, res) => {
+// twitterString : a string/ empty string
+// (test for location known with: if (location || location.twitterString) ... )
+// Better is to test for location specificity
+// location.specficity = 0 => no location
+// location.specficity = 10 => full postcode or lat/long
+// But test, eg: if (!location || location < x) ... since string.prop = undefined and (undefined < x) == false
 app.get('/api', async (req, res) => {
   const { userData, oauthAccessToken, oauthAccessTokenSecret } = req.session
   if (!userData) {
+    console.log(403);
     return res.status(403).send('You are not logged in with Twitter')
   }
   try {
     const followerIds = await twitter.getFollowerIds(userData.screen_name, oauthAccessToken, oauthAccessTokenSecret)
     const matchedIds = await matcher(followerIds)
-    res.send({ total: followerIds.length, matched: matchedIds.length })
+console.log('>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> query:',req.query);
+    let location = [
+      req.query,                         // query: We assume any query passed will include one+ of pc, p7, pc8, latlong
+      req.session.location,              // session: NB Any location object from session may be partial
+      userData.location,                 // userData: is the twitterString, ie location as it appears on user's Twitter profile
+      LocationObject()
+    ];
+    // console.log(location);
+    req.session.location = await populateLocationObject (location, { useGoogle : true} ) ;
+    res.send({ total: followerIds.length, matched: matchedIds.length, location: req.session.location })
   } catch (e) {
+    console.log(e);
     res.status(e.statusCode || 500).send(e.message)
   }
 })
 
+
+// NB For future cleverness with userData.geo_enabled: geo_enabled is deprecated and will always be null. Still available via GET account/settings. This field must be true for the current user to attach geographic data when using POST statuses / update
+app.get('/locationtest', async (req, res) => {
+  const { userData, oauthAccessToken, oauthAccessTokenSecret } = req.session
+  let followerIds = [], matchedIds = [];
+  console.log('Got route /locationtest');
+console.log(req.query);
+  let location = [
+    req.query,                         // query: We assume any query passed will include one+ of pc, p7, pc8, latlong
+    req.session.location
+  ];
+  if (userData) {
+    followerIds = await twitter.getFollowerIds(userData.screen_name, oauthAccessToken, oauthAccessTokenSecret);
+    matchedIds = await matcher(followerIds);
+    location.push (userData.location);
+  }
+  location.push (LocationObject());
+
+  try {
+      req.session.location = await populateLocationObject (location, { useGoogle : true} ) ;
+    } catch (e) {
+      console.log(e);
+      res.status(e.statusCode || 500).send(e.message)
+    }
+
+  // set usingReact to false for express routing (ie, not React):
+  var usingReact = true;
```

Comment:
consider `let` instead of `var`

Reply:
The point of module-scoping it was to draw attention to it, but there is a comment anyway.
Moved to top of file and replaced with `let`